src/main/java/com/auctionapp/api/service/AuthService.java

```diff
@@ -113,6 +74,14 @@ public AuthenticationResponse login(final LoginRequest loginRequest) {
 		String token = jwtProvider.generateToken(authenticate);
 
 		Optional<User> user = userRepository.findByEmail(loginRequest.getEmail());
-		return new AuthenticationResponse(token, loginRequest.getEmail(), user.get().getRole());
+		if (user.get().getStatus() == Status.INACTIVE) {
+			activateUserAccount(user.get());
+		}
```

Comment:
Does this mean that if a deactivated user tries to log in, their account will be reactivated? I'm not sure we want that, especially as there is no prompt or anything, it makes the deactivation feature the same as logging out. This should be removed and checks should be added for a deactivated user not being able to bid, etc. It's not a super high priority, just something to consider when building such features.

Reply:
Yes, it is currently just like logging out in a nutshell, this deactivation feature needs more work definitely.

Should I further work on it in this PR?

Reply:
I think that at the very least a differentiation should be made between logging out and deactivating an account. Disable inactive users from logging into the application and perhaps think about whether you could/would want to delete some of their personal information when you do so without jeopardizing the data the app still needs to display. Other checks can be added at a later date.

Reply:
I have added to login validation to check if the user is deactivated to be disabled from logging again.
And when the deactivation is called, user personal information like first name, last name phone number, date of birth is deleted.